### PR TITLE
Adding missing option for set-chef in both ARM and ASM

### PR DESCRIPTION
**Pull request by Vasu1105**
The content to be added to Changelog is as follows:
Added new options in azure vm extension set-chef command for both ASM and ARM mode
*  --daemon - Configures the chef-client service for unattended execution. The node platform to be Windows. Options: \'none\' or \'service\'. \n \'none\' - Currently prevents the chef-client service from being configured as a service. \n \'service\' - Configures the chef-client to run automatically in the background as a service.
* --chef-service-interval - It specifies the frequency (in minutes) at which the chef-service runs. Pass 0 if you don\'t want the chef-service to be installed on the target machine.
* --secret - The secret key to use to encrypt data bag item values.
* --secret-file  - A file containing the secret key to use to encrypt data bag item values.
* --bootstrap-version - chef-client version to be installed.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@Vasu1105__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: lib/commands/arm/vm/vm._js [hunk 1]
```diff
@@ -632,11 +632,16 @@ exports.init = function (cli) {
       .option('-c, --client-config <client-config>', $('chef client configuration file(i.e client.rb) path'))
       .option('-C, --client-pem <client-pem>', $('chef client pem file path i.e required in validator less bootstrap'))
       .option('-j, --bootstrap-options <bootstrap-json-attribute>', $('Bootstrap options in JSON format. Ex: -j \'{"chef_node_name":"test_node"}\''))
+      .option('-v, --bootstrap-version <number>', $('chef-client version to be installed'))
```
Comment (amarzavery) [inline lib/commands/arm/vm/vm._js]: -v and -vv are reserved switches. Please use something else.

```
help:       -v, --verbose                  use verbose output
help:      -vv                            more verbose with debug output
```

File: lib/commands/asm/vm._js [hunk 2]
```diff
@@ -205,12 +205,16 @@ exports.init = function(cli) {
     .option('-c, --client-config <client-config>', $('chef client configuration file(i.e client.rb) path'))
     .option('-C, --client-pem <client-pem>', $('chef client pem file path i.e required in validator less bootstrap'))
     .option('-j, --bootstrap-options <bootstrap-json-attribute>', $('Bootstrap options in JSON format. Ex: -j \'{"chef_node_name":"test_node"}\''))
-    .option('-v, --bootstrap-version <number>', $('chef-client version to be installed'))
```
Comment (btm) [inline lib/commands/asm/vm._js]: Was `-v` removed because it overlapped another option and didn't work anyway or otherwise conflicted?

Comment (Vasu1105) [inline lib/commands/asm/vm._js]: @btm as mentioned by @amarzavery it's reserved for -verbose so removed it.

**Pull request merged**